### ISSUE TITLE
Same CSS in two different styled react modal

### ISSUE THREAD

**Issue by tvallois** [AlexanderRichey/styled-react-modal#25]
Hi,

First, thanks for the job you do on this library.

We face an issue with it, we have a page with multiple modals in it.
The first:
```javascript
const SequenceModal = Modal.styled`
	background-color: #313131;
	font-size: 1.5 rem;
	line-height: 1;
	padding-left: 20px;
	padding-right: 20px;
	padding-top: 10px;
	padding-bottom: 10px;
	border: 1px solid #0003;
	border-bottom: 25px solid transparent;
	border-radius: .3rem;
	margin-left: 5%;
	margin-right: 5%;
	max-height: 75%;
	max-width: 55rem;
	width: 100%;

	text-align: left;
	position: relative;
	vertical-align: middle;
	display: inline-block;
	overflow: auto;
`;
```
The second:
```javascript
const SConfirmDelete = Modal.styled`
  border: 1px solid #0003;
  border-radius: .3rem;
  margin-left: 5%;
  margin-right: 5%;
  max-height: 85%;
  max-width: 95%;
  width: auto;
  background: #fff;
  color: black;
  transform: translate(0%, -50%);
`;
```
When i trigger the first modal, cancel it then trigger the second one. The second modal has the css attributes of the first one instead of having his own. When i look at the css, it seems that they have both the same className generated by styled-components.
Is it a bug or an expected behavior?

Thanks
```

**Comment by AlexanderRichey** [comment]
Interesting issue. One solution I can think of is to add a unique `key` prop to each modal. Although `key` props are mostly used in lists, they can also be used to mark sibling components as unique and to force rerenders by explicitly telling React that a given component has changed. I think this is worth a shot because, given the behavior you described, it seems possible that React is confused about which modal is which and needs to be told explicitly that they are distinct.

If that doesn't fix it, would you mind posting a link to a [sandbox](https://codesandbox.io/) that reproduces the issue so that I can dig into it a little more?

**Comment by tvallois** [comment]
Your proposition does not fix the issue, i'll make a sandbox asap. Another strange thing, i'm unable to reproduce this issue on my development environment (we use CRA) but it happens 100% in production.

**Comment by tvallois** [comment]
I've created a sandbox however i'm unable to reproduce the issue in it...
https://codesandbox.io/s/autumn-firefly-4d0r9

**Comment by tvallois** [comment]
Ok, the issue seems to be gone if i retrograde styled-components lib version from 5 to 4...

**Comment by AlexanderRichey** [comment]
Thanks for the sandbox. I was able to reproduce the issue by generating the production assets locally. I was also able to fix this issue.

The problem is that `styled-components@5.0.1` has a peer dependency on `react-is@>=16.8`. If you explicitly declare this dependency in your `package.json` (and reinstall your dependencies and rebuild, of course), the modals should behave as expected. Would you mind giving this a shot and letting me know if it works?

I'm going to upgrade `styled-react-modal`'s dependencies and publish a new patch so that others won't run into this issue as well.

**Comment by AlexanderRichey** [comment]
I may have spoken too soon. The real issue seems to be a foolish setting on my end. In the `.babelrc`, which controls `babel-plugin-styled-components`, I had [`ssr`](https://styled-components.com/docs/tooling#serverside-rendering) set to `true`.

My reason for setting `ssr` to `true` was that there could be a case where someone is doing server side rendering and wants a page to be rendered with an open modal. While this is still a possible use-case, I think it's less important than your use-case of having more than one modal per page, which I suspect would be more popular.

I published a new version that fixes the issue and which also upgrades dependencies and does some housekeeping. Thanks for pointing out this issue. :slightly_smiling_face: 

Fixed by [v2.0.1](https://github.com/AlexanderRichey/styled-react-modal/releases/tag/v2.0.1).
